glide-core/src/cluster_scan_container.rs

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+use logger_core::log_info;
+use once_cell::sync::Lazy;
+use redis::{RedisResult, ScanStateRC};
+use sha1_smol::Sha1;
+use std::{collections::HashMap, sync::Mutex};
+
+static CONTAINER: Lazy<Mutex<HashMap<String, ScanStateRC>>> =
```

Comment:
You'll have a memory leak if user starts 100500 scans.
You have to serialize the scan and deserialize it back and store nothing.
A serialized scan could be safely ported from one client to another (if it works vs the same server)

Reply:
user starts 100500 scans at the same time? thats seems very unlikely
 anyhow can you explain:
> You have to serialize the scan and deserialize it back and store nothing.
We storing a ref here, it is not the object itself. 
It is just to hold the scan state object in redis-rs so it wont be dropped. 
When the user will drop the object in the wrapper and the gc will clean it, it will trigger a clean from the table and so it will drop the scan state object in redis-rs since there's no ref hold anywhere. We dont store the state anywhere. its alive as long as the user hold it in the wrapper.

Reply:
If GLIDE is used in a whatever application with long lifetime (e.g. daemonized), number of started/unfinished scans will only grow on time causing a memory consumption. Actually, unclaimed unfinished scans are memory leaks, since you don't define lifetime for them.
To avoid that you need to serialize the scan into a string and return that string instead of the hash.

Reply:
I thought the docs are explaining process. 
I'll explain -
in redis-rs we create an object, we dont return the object, but an RC, a ref counter, so Rust wont drop it.
We cant give the user in the wrapper the ref since its out of the Rust scope and Rust won't know we still hold it and will drop the object.
So the solution is to hold the ref in glide-core inside a table, then it wont be dropped. 
We dont hash something here, the hash is just a random string used as key to the ref. it is not an hashed heavy object, it is just a string, and we store a ref, not an object, it is just an address.
We return the hash to the user wrapped by an object (py implement, which need to be implemented similarly in other languages, already explained the idea to @acarbonetto and it can be implemented the same in java, and i will do the same in node).
The idea of putting it inside an object is that we can add action to the delete function of the object that will be triggered by the GC of each language. I've added a test checking that it is actually works.
When the user drop the object by not referencing to it anymore, which will happen other than if the user hold the object as a static or insert it into statically object, which i dont see any user doing if he dont want to fill is memory 
The del function that triggered removing the ref from the table using the key, and when the ref is removed, Rust will drop the real object.
As long as the user doesn't decide for whatever reason to store statically all his cursors statically somewhere, which of curse is a decision that affect memory consuming, will be no extra memory consumption.
We dont save a ref for a whole scan, it doesnt matter if the user finished the scan or not. If the user drop the cursor theres no object alive anymore.

Reply:
Anyhow - for the simplicity - changed from hash to nanoid